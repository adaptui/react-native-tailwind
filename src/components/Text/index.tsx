import { Text as RNText, TextProps as RNTextProps } from 'react-native';
import { createComponent } from '../../utils/createComponent';
import { createElement } from '../../utils/createElement';

export type LibraryTextProps = {};

export type TextProps = RNTextProps & LibraryTextProps;

export const Text = createComponent<TextProps>(
  (props) => {
    return createElement({
      componentType: RNText,
      props: props,
    });
  },
  { shouldMemo: true }
);
