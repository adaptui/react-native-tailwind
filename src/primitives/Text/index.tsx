import { Text as RNText, TextProps as RNTextProps } from 'react-native';
import type { Dict } from 'src/utils/types';
import { createComponent } from '../../utils/createComponent';

export type LibraryTextProps = Dict<unknown>;

export type TextProps = RNTextProps & LibraryTextProps;

export const Text = createComponent<TextProps>(RNText, { shouldMemo: true });
