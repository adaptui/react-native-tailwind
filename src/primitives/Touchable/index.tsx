import { Pressable, PressableProps } from 'react-native';
import type { Dict } from 'src/utils/types';
import { createComponent } from '../../utils/createComponent';

export type LibraryTouchableProps = Dict<unknown>;

export type TouchableProps = LibraryTouchableProps & PressableProps;

export const Touchable = createComponent<TouchableProps>(Pressable, {
  shouldMemo: true,
});
