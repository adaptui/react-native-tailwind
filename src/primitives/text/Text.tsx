import { Text as RNText } from "react-native";

import {
  As,
  createComponent,
  createElement,
  createHook,
  Options,
  Props,
} from "../../utils/system";

export const useText = createHook<TextOptions>(props => {
  return props;
});

export const Text = createComponent<TextOptions>(props => {
  const htmlProps = useText(props);
  return createElement(RNText, htmlProps);
});

export type TextOptions<T extends As = typeof RNText> = Options<T>;

export type TextProps<T extends As = typeof RNText> = Props<TextOptions<T>>;
