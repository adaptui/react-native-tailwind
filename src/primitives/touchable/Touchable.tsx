import { Pressable, PressableProps } from "react-native";

import { createComponent } from "../../utils/createComponent";
import type { Dict } from "../../utils/types";

export type LibraryTouchableProps = Dict<unknown>;

export type TouchableProps = LibraryTouchableProps & PressableProps;

export const Touchable = createComponent<TouchableProps>(Pressable, {
  shouldMemo: true,
});
