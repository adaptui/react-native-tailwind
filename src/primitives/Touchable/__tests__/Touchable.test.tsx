import { render } from '@testing-library/react-native';
import React from 'react';
import { Touchable } from '../../Touchable';
import { Text } from '../../Text';

describe('basic', () => {
  it('should render correctly', () => {
    const component = render(
      <Touchable>
        <Text>Hello, World</Text>
      </Touchable>
    );
    expect(component).toMatchSnapshot();
  });

  it('should assign a ref', () => {
    const ref = React.createRef();
    render(
      <Touchable ref={ref}>
        <Text>Hello, World</Text>
      </Touchable>
    );
    expect(ref.current).toMatchSnapshot();
  });

  it('should render correctly with a style prop', () => {
    const component = render(
      <Touchable
        style={{
          padding: 4,
          margin: 10,
          backgroundColor: 'red',
        }}
      >
        <Text>Hello, World</Text>
      </Touchable>
    );
    expect(component).toMatchSnapshot();
  });

  it('should render children correctly with blue background', () => {
    const component = render(
      <Touchable
        style={{
          padding: 4,
          margin: 10,
          backgroundColor: 'blue',
        }}
      >
        <Text> This is a Touchable </Text>
      </Touchable>
    );
    expect(component).toMatchSnapshot();
  });
});
