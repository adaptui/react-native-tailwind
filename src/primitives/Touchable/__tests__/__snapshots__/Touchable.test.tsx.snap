// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic should assign a ref 1`] = `null`;

exports[`basic should render children correctly with blue background 1`] = `
<View
  accessible={true}
  collapsable={false}
  elementRef={null}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "blue",
      "margin": 10,
      "padding": 4,
    }
  }
>
  <Text
    elementRef={null}
  >
     This is a Touchable 
  </Text>
</View>
`;

exports[`basic should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  elementRef={null}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <Text
    elementRef={null}
  >
    Hello, World
  </Text>
</View>
`;

exports[`basic should render correctly with a style prop 1`] = `
<View
  accessible={true}
  collapsable={false}
  elementRef={null}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "red",
      "margin": 10,
      "padding": 4,
    }
  }
>
  <Text
    elementRef={null}
  >
    Hello, World
  </Text>
</View>
`;
